activemq-broker/src/main/java/org/apache/activemq/broker/scheduler/Job.java

```diff
@@ -78,4 +80,6 @@ public interface Job {
      */
     public int getExecutionCount();
 
+    public abstract ActiveMQDestination getDest();
```

Comment:
Let's change this to just be a 'String' type that we can specify as the destination.getPhysicalName()

Reply:
Something like this? 

getAllJobs JMX operation:

```diff

public class JobSchedulerView implements JobSchedulerViewMBean {

   public TabularData getAllJobs(String startTime, String finishTime) throws Exception {
         OpenTypeFactory factory = OpenTypeSupport.getFactory(Job.class);
         CompositeType ct = factory.getCompositeType();
         TabularType tt = new TabularType("Scheduled Jobs", "Scheduled Jobs", ct, new String[] { "jobId" });
         TabularDataSupport rc = new TabularDataSupport(tt);
         long start = JobSupport.getDataTime(startTime);
         long start = JobSupport.getDataTime(startTime);
         long finish = JobSupport.getDataTime(finishTime);
         List<Job> jobs = this.jobScheduler.getAllJobs(start, finish);
+        OpenWireFormat wireFormat = new OpenWireFormat();
         for (Job job : jobs) {
+            Message msg = (Message) wireFormat.unmarshal(new ByteSequence(job.getPayload()));
+            ActiveMQDestination destination = (ActiveMQDestination) msg.getJMSDestination();
+            job.setDestinationName(destination.getPhysicalName());
             rc.put(new CompositeDataSupport(ct, factory.getFields(job)));
         }
         return rc;

```

Reply:
Yep!